pkg/controller/direct/mappings/assertions.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mappings
+
+import "k8s.io/klog/v2"
+
+// AssertEnabled is true if we are running with assertions enabled.
+// Like traditional assertions, assertions are runtime checks that
+// will panic the program, but we only run those checks if assertion
+// checks are enabled.
+var AssertEnabled bool = true
```

Comment:
Not convinced that bool is the correct type here. I'm betting we end up wanting an enumeration for things like testing only assertion and production assertions.

Reply:
Agreed, bool is just all I needed right now!